### PR TITLE
docs: minor improvments/fixes

### PR DESCRIPTION
**Pull request by KevFan**
* Fix some typos
* Fix broken references to other docs

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -359,28 +359,28 @@ All the identity verification methods supported by Authorino can be configured r
...
+A typical use-case for this feature is token normalization. Say you have more than one identity source listed in your `AuthConfig` but each source issues an access token with a different JSON structure – e.g. two OIDC issuers that use different names for custom JWT claims of similar meaning; when two different identity verification/authentication methods are combined, such as API keys (whose identity objects are the corresponding Kubernetes `Secret`s) and Kubernetes tokens (whose identity objects are Kubernetes UserInfo data).
 
-In such cases, identity extension can be used to normalize the token so it always includes the same set of JSON properties of interest, regardless of the source of identity that issued the original token verified by Authorino. This simplifies the writing of authorization policies and configuration of dynamic responses.
+In such cases, identity extension can be used to normalize the token to always includes the same set of JSON properties of interest, regardless of the source of identity that issued the original token verified by Authorino. This simplifies the writing of authorization policies and configuration of dynamic responses.
```
Comment (guicassolato) [inline docs/features.md:375]:
```suggestion
In such cases, identity extension can be used to normalize the token to always include the same set of JSON properties of interest, regardless of the source of identity that issued the original token verified by Authorino. This simplifies the writing of authorization policies and configuration of dynamic responses.
```

File: docs/features.md [hunk 2]
```diff
@@ -911,7 +911,7 @@ spec:
   - name: metadata-source
     http:
       endpoint: https://my-metadata-source.io
-    when: # only fetch the external metadata if the context is HTTP method different than OPTIONS
+    when: # only fetch the external metadata if the context is HTTP method different other than OPTIONS
```
Comment (guicassolato) [inline docs/features.md:914]:
```suggestion
    when: # only fetch the external metadata if the context is HTTP method other than OPTIONS
```

**Review by guicassolato — DISMISSED** [review]
😳 We need to review the use of [misspell](https://github.com/Kuadrant/authorino/blob/b0101ed8ae328dee3a8ef85768bd0d79cf0cc81e/.github/workflows/code-style.yaml#L131).

**Comment by KevFan** [comment]
> 😳 We need to review the use of [misspell](https://github.com/Kuadrant/authorino/blob/b0101ed8ae328dee3a8ef85768bd0d79cf0cc81e/.github/workflows/code-style.yaml#L131).

Oh, I didn't realise we used misspell in github actions. I've created https://github.com/Kuadrant/authorino/issues/415 to track this 👍

**Review by guicassolato — APPROVED** [review]

**Pull request merged**